core/src/main/scala/org/http4s/headers/Server.scala

```diff
@@ -17,15 +17,23 @@
 package org.http4s
 package headers
 
-import org.http4s.parser.HttpHeaderParser
+import cats.parse._
 import org.http4s.util.Writer
 
 object Server extends HeaderKey.Internal[Server] with HeaderKey.Singleton {
   def apply(id: ProductId): Server =
     new Server(id, Nil)
 
   override def parse(s: String): ParseResult[Server] =
-    HttpHeaderParser.SERVER(s)
+    ParseResult.fromParser(parser, "invalid Server")(s)
+
+  private[http4s] val parser = {
```

Comment:
`User-Agent` and Server are very similar, maybe they should share the parser up to the last map?

Reply:
moved them to the `AdditionalParsers` if you have a better location in mind please let me know
https://github.com/http4s/http4s/pull/4208/commits/b89f9d6ebfbea75f7c51749d262d6361146b3d44